// @appcues/component-library/dist/CCheckbox.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _chunkB2edacfb = require("./chunk-b2edacfb.js");

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _CheckboxGroup = _interopRequireDefault(require("./CheckboxGroup.js"));

var _CheckboxIndicator = _interopRequireDefault(require("./CheckboxIndicator.js"));

var _type = require("./type.js");

require("prop-types");

require("./SelectionGroup.js");

require("./CFontIcon.js");

require("./layout.js");

require("./chunk-b15f6972.js");

require("./chunk-6104512b.js");

require("./chunk-76875e58.js");

require("./chunk-cb015231.js");

require("./chunk-f2932d1d.js");

require("./chunk-eb19ede6.js");

require("./chunk-8219403d.js");

require("./chunk-cfb1dc74.js");

require("./chunk-e8276423.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject3() {
  var data = (0, _chunkB2edacfb.a)(["\n  display: flex;\n  align-items: flex-start;\n  margin-bottom: 10px;\n  max-width: 400px;\n  font-weight: 600;\n  color: ", ";\n  cursor: pointer;\n\n  &:first-of-type {\n    margin-top: 6px;\n  }\n\n  &:hover {\n    color: ", ";\n\n    .checkbox {\n      box-shadow: inset 0 0 0 ", "\n        ", ";\n    }\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _chunkB2edacfb.a)(["\n  flex-direction: column;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _chunkB2edacfb.a)(["\n  position: absolute;\n  top: 0;\n  left: 0;\n  opacity: 0;\n  z-index: -1;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var CCheckbox = function CCheckbox(_ref) {
  var children = _ref.children,
      label = _ref.label,
      caption = _ref.caption,
      props = (0, _chunkB2edacfb.j)(_ref, ["children", "label", "caption"]);
  return _react.default.createElement(StyledLabel, {
    checked: props.checked
  }, _react.default.createElement(HiddenInput, (0, _chunkB2edacfb.g)({
    type: "checkbox"
  }, props)), _react.default.createElement(_CheckboxIndicator.default, {
    isChecked: props.checked
  }), _react.default.createElement(Content, null, _react.default.createElement(_type.Text, null, label), children, caption && _react.default.createElement(_type.Caption, null, caption)));
};

CCheckbox.displayName = "CCheckbox";
CCheckbox.Group = _CheckboxGroup.default;

var HiddenInput = _styledComponents.default.input.withConfig({
  displayName: "CCheckbox__HiddenInput",
  componentId: "sc-1nvbulx-0"
})(_templateObject());

var Content = _styledComponents.default.div.withConfig({
  displayName: "CCheckbox__Content",
  componentId: "sc-1nvbulx-1"
})(_templateObject2());

var StyledLabel = _styledComponents.default.label.withConfig({
  displayName: "CCheckbox__StyledLabel",
  componentId: "sc-1nvbulx-2"
})(_templateObject3(), function (props) {
  return props.theme["$gray-7"];
}, function (props) {
  return props.theme["$blue-dark"];
}, function (props) {
  return props.checked ? "8px" : "2px";
}, function (props) {
  return props.theme["$blue-dark"];
});

CCheckbox.__docgenInfo = {
  "description": "",
  "methods": [],
  "displayName": "CCheckbox"
};
var _default = CCheckbox;
exports.default = _default;
CCheckbox.__docgenInfo = {
  "description": "",
  "methods": [],
  "displayName": "CCheckbox"
};
