// @appcues/component-library/dist/CMenu.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _chunkB2edacfb = require("./chunk-b2edacfb.js");

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _classnames = _interopRequireDefault(require("classnames"));

var _chunk3b2cded = require("./chunk-3b2cded6.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject() {
  var data = (0, _chunkB2edacfb.a)(["\n  background-color: white;\n  position: absolute;\n  padding: 16px 0;\n  margin-top: 0px;\n  list-style: none;\n  border-radius: 4px;\n  overflow-y: auto;\n  box-shadow: ", ";\n  z-index: 3;\n  width: ", ";\n  padding: 0;\n  cursor: default;\n  &.right {\n    left: calc(100% + 3px);\n  }\n  &.up {\n    bottom: 3px;\n  }\n  &.down {\n    top: 3px;\n  }\n  &.left {\n    right: calc(100% + 3px);\n  }\n  .separator {\n    margin: 0;\n    width: 100%;\n    height: 1px;\n    color: ", ";\n  }\n\n  .option-detail-container {\n    display: block;\n    box-sizing: border-box;\n    padding: 9px 16px;\n    cursor: default;\n  }\n  > li {\n    &:focus-within {\n      .list-row-item {\n        background-color: ", ";\n      }\n    }\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var CMenu = function CMenu(_ref) {
  var className = _ref.className,
      children = _ref.children,
      id = _ref.id;
  return _react.default.createElement(Menu, {
    className: (0, _classnames.default)("menu", "up", "right", "cmenu", className),
    onClick: function onClick(e) {
      e.stopPropagation();
    },
    role: "menu",
    id: id
  }, _react.default.Children.map(children, function (child, i) {
    return _react.default.createElement("li", {
      key: i,
      tabIndex: "-1"
    }, child);
  }));
};

var Menu = _styledComponents.default.ul.withConfig({
  displayName: "CMenu__Menu",
  componentId: "cf3ebp-0"
})(_templateObject(), function (props) {
  return props.theme["$box-shadow-3"];
}, function (props) {
  return props.menuWidth ? props.menuWidth + "px" : "256px";
}, function (props) {
  return props.theme["$dark-tint-10"];
}, function (props) {
  return props.theme["$gray-2"];
});

CMenu.Separator = function () {
  return _react.default.createElement("hr", {
    className: "separator"
  });
};

CMenu.Separator.displayName = "Separator";
CMenu.displayName = "CMenu";
CMenu.__docgenInfo = {
  "description": "",
  "methods": [],
  "displayName": "CMenu"
};
var _default = CMenu;
exports.default = _default;
CMenu.__docgenInfo = {
  "description": "",
  "methods": [],
  "displayName": "CMenu"
};
